src/game_display.cpp

```diff
@@ -664,3 +676,98 @@ display::overlay_map& game_display::get_overlays()
 {
 	return overlay_map_;
 }
+
+std::vector<texture> game_display::get_reachmap_images(const map_location& loc, image::TYPE image_type)
+{
+	std::vector<std::string> names;
+	const auto adjacent = get_adjacent_tiles(loc);
+
+	enum visibility { REACH = 0, CLEAR = 1 };
+	std::array<visibility, 6> tiles;
+
+	const std::array image_prefix{&game_config::reach_map_prefix};
+	DBG_DP << "Loaded image prefix: " << game_config::reach_map_prefix;
+
+	for(int i = 0; i < 6; ++i) {
+		if(reach_map_.find(adjacent[i]) != reach_map_.end()) {
+			DBG_DP << "Adjacent " << std::to_string(i) << " to tile " << loc << " is REACHABLE";
+			tiles[i] = REACH;
+		} else {
+			DBG_DP << "Adjacent " << std::to_string(i) << " to tile " << loc << " is NOT REACHABLE";
+			tiles[i] = CLEAR;
+		}
+	}
+
+	for(int v = REACH; v != CLEAR; ++v) {
+		// Find somewhere that doesn't have overlap to use as a starting point
+		int start;
+		for(start = 0; start != 6; ++start) {
+			if(tiles[start] != v) {
+				break;
+			}
+		}
+
+		if(start == 6) {
+			// Completely surrounded by reach. This may have a special graphic.
+			DBG_DP << "Tried completely surrounding";
+			const std::string name = *image_prefix[v] + "-all.png";
+			if(image::exists(name)) {
+				names.push_back(name);
+				continue;
+			}
+			// No special graphic found. We'll just combine some other images
+			// and hope it works out.
+			start = 0;
+		}
+
+		// always push the background image, as the graphics are meant to have it
+		const std::string name = *image_prefix[v] + ".png";
```

Comment:
Doesn't need to be its own variable. Just do `push_back(*image_prefix[v] + ".png")`